### ISSUE TITLE
Support delete by query

### ISSUE THREAD

**Issue by kelaban** [karmi/retire#771]
Forgive me if this already available. I think delete by query would be a great addition to the tire api. It should be fairly straight forward as it uses same search query dsl.

For example

``` ruby
Model.delete do
  filter :terms, :tags => ['ruby']
end
```

**Comment by karmi** [comment]
The problem is that RestClient, the default HTTP client for Tire, does not support sending body with `DELETE`, see: https://github.com/rest-client/rest-client/blob/master/lib/restclient.rb#L83-L85

**Comment by phoet** [comment]
@karmi do you think it would make sense to add such a method to RestClient? it would be fairly trivial to add a `delete_with_payload` method (or just add it to tire).

``` ruby
  def self.delete_with_payload(url, payload, headers={}, &block)
    Request.execute(:method => :delete, :url => url, :payload => payload, :headers => headers, &block)
  end
```

**Comment by karmi** [comment]
Yeah, absolutely, but it would be better to add it in upstream to RestClient directly... and possibly monkey patch the same code on our side for keeping compatibility with low RestClient versions, if needed.

**Comment by phoet** [comment]
i will try to take care of this over the weekend

**Comment by karmi** [comment]
As noted in #309,  `Curb` also doesn't support `DELETE` request with a body, so serializing the payload into the `source` URL parameter seems like most convenient and safe way.

**Comment by phoet** [comment]
just a note: http://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request

**Comment by karmi** [comment]
Closed by 55d12c0caaf09f8084266b08719e47de8fef2919
